### DISCUSSION TITLE
20240411 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#89] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - 
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x
- Mick van Dijke(@mickvandijke) - 
- Iván Santana(@ivan9989-pkj) - x
- Ismael Torrado(@Lepisma07) -
- Borja (@BorjaPractica) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Testing in Rust

New article: [Static vs Dynamic mocks in Rust](https://github.com/nautilus-cyberneering/testing-in-rust/blob/main/docs/static-vs-dynamic-mocks-in-rust.md)

### Torrust Tracker

PR review:

- https://github.com/torrust/torrust-tracker/pull/782

Issues:

- [x] https://github.com/torrust/torrust-tracker/issues/777
- [x] https://github.com/torrust/torrust-tracker/issues/565

I also reviewed some old issues regarding timeouts:

- https://github.com/torrust/torrust-tracker/issues/612
- https://github.com/torrust/torrust-tracker/issues/613

I've opened an issue in the axum-server repo:

https://github.com/programatik29/axum-server/issues/116

They supposedly added a feature to allow setting a timeout, but it seems it was removed. I guess there must be a way to do it directly now.

### Torrust Index

PR Review: 

- https://github.com/torrust/torrust-index/pull/557

I also discussed with @mario-nt how to implement the authorization service.

### Torrust Index GUI

Discussion: https://github.com/torrust/torrust-index-gui/discussions/519

### Learning

I've finished reading the Rust book (no starch press)

**Comment by mario-nt** [comment]
### Index
	
- https://github.com/torrust/torrust-index/issues/448

### Training/Research

- Tidy first? book
- Role authorization research
- https://github.com/nautilus-cyberneering/testing-in-rust/blob/main/docs/static-vs-dynamic-mocks-in-rust.md

### New Issues

- https://github.com/torrust/torrust-index/issues/556

**Comment by grmbyrn** [comment]
## Rust
* [Rust Guessing Game](https://github.com/grmbyrn/rust-guessing-game)
    * Strings
    * Mutable reference
    * If expression
    * Match
    * Loops
    * Vectors
    * For in loop

* [Rust Game](https://github.com/grmbyrn/rust-game)
    * Enums
    * Tuples
    * Creating an array
    * Looping with arrays
    * Implementing traits for enums
    * Derivable traits
    * Getting input from the user

## Torrust
* Setting up the Torrust tracker

## MERN project
* Revising aspects of MERN in [this project](https://github.com/grmbyrn/technotes) to improve backend/fullstack skills.

**Comment by josecelano** [comment]
Some resources shared during the meeting to learn Rust:

- https://www.youtube.com/watch?v=jAm7xrRxEUE&list=PL_c9BZzLwBRIymgB73pHXIgazpB-uszKU&index=3        
- https://github.com/rust-lang/rustlings
- https://doc.rust-lang.org/book/s
- https://doc.rust-lang.org/rust-by-example/
